packages/workers/src/controllers/scan-batch-request-feed-controller.ts

```diff
@@ -0,0 +1,99 @@
+import { ServiceConfiguration } from 'common';
+import { inject, injectable } from 'inversify';
+import { Logger } from 'logger';
+import { OnDemandPageScanRunResultProvider, PageScanRequestProvider, PartitionKeyFactory, WebController } from 'service-library';
+import {
+    ItemType,
+    OnDemandPageScanBatchRequest,
+    OnDemandPageScanRequest,
+    OnDemandPageScanResult,
+    PartitionKey,
+    ScanRunBatchRequest,
+} from 'storage-documents';
+
+// tslint:disable: no-any
+
+@injectable()
+export class ScanBatchRequestFeedController extends WebController {
+    public readonly apiVersion = '1.0';
+    public readonly apiName = 'scan-batch-request-feed';
+
+    public constructor(
+        @inject(cosmosContainerClientTypes.OnDemandScanBatchRequestsCosmosContainerClient)
+        private readonly cosmosContainerClient: CosmosContainerClient,
+        @inject(OnDemandPageScanRunResultProvider) private readonly onDemandPageScanRunResultProvider: OnDemandPageScanRunResultProvider,
+        @inject(PageScanRequestProvider) private readonly pageScanRequestProvider: PageScanRequestProvider,
+        @inject(PartitionKeyFactory) private readonly partitionKeyFactory: PartitionKeyFactory,
+        @inject(ServiceConfiguration) protected readonly serviceConfig: ServiceConfiguration,
+        @inject(Logger) protected readonly logger: Logger,
+    ) {
+        super();
+    }
+
+    public async handleRequest(...args: any[]): Promise<void> {
+        const batchDocuments = <OnDemandPageScanBatchRequest[]>args[0];
+        await Promise.all(
+            batchDocuments.map(async document => {
+                await this.processDocument(document);
+            }),
+        );
+    }
+
+    protected validateRequest(...args: any[]): boolean {
+        const batchDocuments = <OnDemandPageScanBatchRequest[]>args[0];
+
+        return this.validateRequestData(batchDocuments);
+    }
+
+    private async processDocument(batchDocument: OnDemandPageScanBatchRequest): Promise<void> {
+        const requests = batchDocument.scanRunBatchRequest.filter(request => request.scanId !== undefined);
+        if (requests.length > 0) {
+            await this.writeRequestsToQueueContainer(requests);
+            await this.writeRequestsToPermanentContainer(requests);
+            await this.cosmosContainerClient.deleteDocument(batchDocument.id, batchDocument.partitionKey);
+        }
+    }
+
+    private async writeRequestsToPermanentContainer(requests: ScanRunBatchRequest[]): Promise<void> {
+        const requestDocuments = requests.map<OnDemandPageScanResult>(request => {
+            return {
+                id: request.scanId,
+                url: request.url,
+                priority: 0,
+                itemType: ItemType.onDemandPageScanRunResult,
+                partitionKey: this.partitionKeyFactory.createPartitionKeyForDocument(ItemType.onDemandPageScanRunResult, request.scanId),
```

Comment:
it might be better if the data service class takes the ownership of  generating system related props like partition key, documentType etc.

Reply:
wan't fix, I did refactor you code out to factory for creating partition key that is sufficient for the current scenario, rest is simple object creation w/o any complex logic that need any wrappers